StarRailBattleSim/src/relics/relics/HunterOfTheGlacialForest.java

```diff
@@ -0,0 +1,34 @@
+package relics.relics;
+
+import characters.AbstractCharacter;
+import enemies.AbstractEnemy;
+import powers.PermPower;
+import powers.PowerStat;
+import powers.TempPower;
+import relics.AbstractRelicSetBonus;
+
+import java.util.ArrayList;
+
+public class HunterOfTheGlacialForest extends AbstractRelicSetBonus {
+    public HunterOfTheGlacialForest(AbstractCharacter owner) {
+        super(owner);
+    }
+
+    public HunterOfTheGlacialForest(AbstractCharacter owner, boolean fullSet) {
+        super(owner, fullSet);
+    }
+
+    @Override
+    public void onEquip() {
+        if (this.owner.elementType == AbstractCharacter.ElementType.ICE) {
+            this.owner.addPower(PermPower.create(PowerStat.SAME_ELEMENT_DAMAGE_BONUS, 10, "Hunter of the Glacial Forest Ice Boost"));
+        }
+    }
+
+    @Override
+    public void afterAttackFinish(AbstractCharacter character, ArrayList<AbstractEnemy> enemiesHit, ArrayList<AbstractCharacter.DamageType> types) {
```

Comment:
would also likely need an `onAfterUseUltimate` hook to work properly. Also I think the power would need `justApplied` set to true so it doesn't decrement immediately.